pkg/proxy/proxy.go

```diff
@@ -215,7 +214,18 @@ retryCreatePort:
 			redir.implementation, err = createKafkaRedirect(redir, kafkaConfiguration{}, DefaultEndpointInfoRegistry)
 
 		case policy.ParserTypeHTTP:
-			redir.implementation, err = createEnvoyRedirect(redir, p.stateDir, p.XDSServer, wg)
+			redir.implementation, err = createEnvoyRedirect(redir, p.stateDir, p.XDSServer, wg, func() (uint16, error) {
+				// Reallocate proxyport after a NACK has been received.
+				p.mutex.Lock()
+				newPort, err := p.allocatePort()
+				if err == nil {
```

Comment:
You must return `err` if not `nil`. Right now, in case of error, you return an unspecified `newPort` value and a `nil` error.

Reply:
Fixed, thanks for spotting this!